test/common/testlib.py

```diff
@@ -743,27 +761,30 @@ def setUp(self):
         if opts.address and self.provision is not None:
             raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
 
-        self.machine = None
         self.browser = None
         self.machines = {}
         provision = self.provision or {'machine1': {}}
 
-        # First create all machines, wait for them later
-        for key in sorted(provision.keys()):
-            options = provision[key].copy()
-            if 'address' in options:
-                del options['address']
-            if 'dns' in options:
-                del options['dns']
-            if 'dhcp' in options:
-                del options['dhcp']
-            machine = self.new_machine(**options)
-            self.machines[key] = machine
-            if not self.machine:
+        test_method = getattr(self.__class__, self._testMethodName)
+        if getattr(test_method, "_testlib__non_destructive", False) and not opts.address:
+            self.machines['machine1'] = MachineCase.get_global_machine()
+            self.machine = MachineCase.get_global_machine()
+        else:
```

Comment:
The else branch would now formally run if `opts.address` is True. I just wonder how that case was handled in `setUp()` in master, and how it is handled here?

Reply:
the opts.address case is handled in new_machine (as before).